packages/liveblocks-react-ui/src/utils/use-controllable-state.ts

```diff
@@ -2,11 +2,29 @@ import { console } from "@liveblocks/core";
 import { useCallback, useEffect, useRef, useState } from "react";
 
 export function useControllableState<T>(
+  /**
+   * The value used as an uncontrolled fallback if neither `value` nor `defaultValue` are provided.
+   */
+  fallbackValue: T,
+
+  /**
+   * The controlled value.
+   */
   value?: T,
+
+  /**
+   * The event handler called when the value changes, if the value is controlled.
+   */
   onChange?: (value: T) => void,
+
+  /**
+   * The default uncontrolled value.
+   */
   defaultValue?: T
```

Comment:
Could it suffice to make `defaultValue` mandatory and remove `fallbackValue`?

- Uncontrolled: (only) use a `defaultValue`, no `value`
- Controlled: have a `defaultValue`, and a `value` (but `defaultValue` would not get used unless `value` gets set to `undefined` at a later time)

Reply:
I thought that it would mean giving up on the console warning because we wouldn't be able to differentiate between a user-set value and a default one, but the warning is based on if `value` changes so yeah, definitely possible. Done in https://github.com/liveblocks/liveblocks/pull/2449/commits/9e4fa022b9a95b69fa062e28c15f0d3e08c77304.